.github/workflows/pdm-lock-automation.yaml

```diff
@@ -76,17 +92,39 @@ jobs:
           else
             echo "Virtual environment already exists."
           fi
+
           source .venv/bin/activate
+
+          echo "Generating/Updating pdm.lock..."
           pdm lock -G :all -v
+
+          echo "Stashing changes if there are any..."
+          git stash || echo "No stash entries found, continuing..."
+
+          echo "Pulling remote ${{ github.head_ref }} branch..."
+          git pull origin ${{ github.head_ref }} --allow-unrelated-histories --no-rebase
+
+          echo "Trying to pop stash if any..."
+          if git stash list | grep -q 'stash@{0}'; then
+            echo "Popping stash..."
+            git stash pop || echo "Failed to pop stash. Continue with other changes."
+          else
+            echo "No stash found, skipping pop."
+          fi
+
           if git diff --quiet; then
             echo "No changes between the current branch and the ${{ github.head_ref }} branch."
             echo "Nothing to commit"
           else
-            git pull origin ${{ github.head_ref }} --allow-unrelated-histories --no-rebase
             echo "Changes detected between the current branch and the ${{ github.head_ref }} branch."
+
+            echo "Adding pdm.lock changes..."
             git add pdm.lock
-            pip install pre-commit~=3.6.2
-            pre-commit run pdm-lock-check
+
+            echo "Running pre-commit pdm hook..."
+            pdm lock --check
```

Comment:
@kirtimanmishrazipstack Can you please move this before `git add`?

Reply:
done @hari-kuriakose